### PR TITLE
feat: support JS files and babel parsers

### PR DESCRIPTION
**Pull request by duniul**
Fixes https://github.com/simonhaenisch/prettier-plugin-organize-imports/issues/7.

Couldn't find any contribution guidelines or anything, so I've just tried to replicate the conventions and patterns I could see.

## Context
This feels like the best opinionated import sorter plugin for Prettier, but my team needs it to work with both JS and TS! It also doesn't work for TypeScript files when using the `babel-ts` parser at the moment.

I saw https://github.com/simonhaenisch/prettier-plugin-organize-imports/issues/7 and thought I'd take a stab. We've been using these changes locally for a while now and haven't had any issues 👍

### Features
- adds support for JS files without a `tsconfig.json` by always treating `allowJs` as `true` (doesn't affect the actual import sorting, just enables it for JS files)
- allows organizing imports when using the `babel` and `babel-ts` parsers too

### Tests
- test suite now runs for a dynamic set of prettier options (only tests the supported parsers for now)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "prettier-plugin-organize-imports",
-  "version": "1.1.1",
+  "version": "1.2.1",
```
Comment (simonhaenisch) [inline package.json:3]: Please undo this change, version will be bumped when I actually do a release (also it would be `1.2.0`).

File: index.js [hunk 2]
```diff
@@ -88,11 +91,22 @@ class ServiceHost {
...
+/**
+ * Sets `organizeImports` as the given parsers preprocess step, or merges it with the existing one.
+ *
+ * @param {import('prettier').Parser} parser prettier parser
+ */
+const withOrganizeImportsPreprocess = (parser) => {
+	return {
+		...parser,
+		preprocess: parser.preprocess 
+			? (text, options) => organizeImports(babelParsers['babel-ts'].preprocess(text, options), options)
```
Comment (simonhaenisch) [inline index.js:103]:
```suggestion
			? (text, options) => organizeImports(parser.preprocess(text, options), options) 
```

**Review by simonhaenisch — CHANGES_REQUESTED** [review]
Thanks for the PR (:

I've added a couple comments. I yet have to look at the changes to the tests properly before this can go ahead though. Will try to find some time soon.

File: package.json (hunk 1 above)
Comment (duniul) [reply package.json]: Done!

File: index.js (hunk 2 above)
Comment (duniul) [reply index.js]: 🙈 Remains from a messy rebase. Fixed now!

**Comment by duniul** [comment]
> I yet have to look at the changes to the tests properly before this can go ahead though.

@simonhaenisch 
No stress! Diff looks messy, but the tests are untouched other than that they run for each of the parsers.

**Comment by simonhaenisch** [comment]
> the tests are untouched other than that they run for each of the parsers

yeah that's what it looked like but i just want to have a good look first 🤓

File: test.js [hunk 3]
```diff
@@ -6,78 +6,83 @@ const prettier = require('prettier');
...
+const prettify = (code, options) => prettier.format(code, { plugins: ['.'], filepath: 'file.ts', ...options });
 
-test('sorts imports', (t) => {
-	const code = `
-		import { foo, bar } from "foobar"
+const prettierOptions = [{ parser: 'babel' }, { parser: 'babel-ts' }, { parser: 'typescript' }];
 
-		export const foobar = foo + bar
-	`;
+prettierOptions.forEach((options) => {
```
Comment (simonhaenisch) [inline test.js:13]: Hm, wondering whether this is a good use case for a "macro" 🤔 

https://github.com/avajs/ava/blob/main/docs/01-writing-tests.md#reusing-test-logic-through-macros

Comment (duniul) [reply test.js:13]: I wondered that too, but I'm not very familiar with Ava. The documentation wasn't really clear about whether or not to use macros for multiple tests at a time, or if they were just for reusing individual tests.

With macros you'd still have to write one line per test + test case without a loop (so 18 lines here).

I based my approach on the thread here: https://github.com/avajs/ava/issues/1967.

**Review by simonhaenisch — APPROVED** [review]
I just refactored the tests to use macros, seems to work pretty well. Afaik there's a benefit for static analysis when the `test` calls are not defined inside loops (otherwise it might go unnoticed that some tests are not actually running).

**Comment by simonhaenisch** [comment]
Thanks again for the contribution! Will just do a couple more things (like updating the readme), then I'll publish to npm 🥳

**Comment by simonhaenisch** [comment]
Hm I just noticed that `prettier/parser-babel` is not available in Prettier v1 (it was called `parser-babylon` before). Therefore I'll have to bump the `prettier` peer dependency from `>=1.10` to `>=2.0` which will actually be a breaking change :/ Oh well, maybe it's just time for version 2 of this plugin then.

**Comment by simonhaenisch** [comment]
Just released `2.0.0` 🎉 🙌
